client/src/main/java/com/lightbend/lagom/javadsl/client/AkkaCircuitBreakingServiceLocator.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package com.lightbend.lagom.javadsl.client;
+
+import com.lightbend.lagom.internal.client.CircuitBreakers;
+import com.lightbend.lagom.javadsl.api.CircuitBreaker;
+import com.lightbend.lagom.javadsl.api.Descriptor;
+import com.lightbend.lagom.javadsl.api.ServiceLocator;
+
+import java.net.URI;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.function.Function;
+
+/**
+ * Abstract service locator that provides circuit breaking using Akka.
+ *
+ * Generally, only the {@link #locate(String, Descriptor.Call)} method needs to be implemented, however
+ * {@link #doWithServiceImpl(String, Descriptor.Call, Function)} can be overridden if the service locator wants to
+ * handle failures in some way.
+ */
+public abstract class AkkaCircuitBreakingServiceLocator implements ServiceLocator {
```

Comment:
I don't see anything Akka specific in this class (in fact there are no imports to the akka package), so why the name? Yes, the `CircuitBreakers` uses Akka, and I guess one could extract an interface so that something different than Akka could be used, no?

Basically, I'm just wondering why `CircuitBreakingServiceLocator` isn't a better name.